tests/test_resolvers/test_sub.py

```diff
@@ -0,0 +1,31 @@
+from unittest.mock import Mock
+
+from sceptre.resolvers import Resolver
+from sceptre.resolvers.sub import Sub
+
+
+class FirstResolver(Resolver):
+    def resolve(self):
+        return "first"
+
+
+class SecondResolver(Resolver):
+    def resolve(self):
+        return "second"
+
+
+class TestSub:
+    def test_resolve__combines_resolvers_into_single_string(self):
+        argument = [
+            "{first} is {first_value}; {second} is {second_value}",
+            {
+                "first": FirstResolver(),
+                "second": SecondResolver(),
+                "first_value": 123,
+                "second_value": 456,
+            },
```

Comment:
how about a test when one of the resolvers returns a non-string (list or dict) object?

how about a test with a missing replacement parameter? 
```
        argument = [
            "{first} and {second}",
            {
                "second": "B"
            },
``` 

How about a test where resolver returns `None`?
```
        argument = [
            "{first} and {second}",
            {
                "first": ResolverReturnsNone(),
                "second": "B"
            },
```

Reply:
Additional validation and tests have been added.